datagateway_api/src/common/filter_order_handler.py

```diff
@@ -99,6 +108,29 @@ def add_icat_relations_for_non_related_fields_of_panosc_related_entities(
                 python_icat_include_filter = PythonICATIncludeFilter(icat_relations)
                 self.filters.append(python_icat_include_filter)
 
+    def add_icat_relations_for_panosc_non_related_fields(
+        self, panosc_entity_name,
+    ):
+        """
+        Retrieve ICAT relations and create a `PythonICATIncludeFilter` for these ICAT
+        relations
+
+        :param panosc_entity_name: A PaNOSC entity name e.g. "Dataset"
+        :type panosc_entity_name: :class:`str`
+        :return: ICAT relations for the non related fields of the given PaNOSC entity
+        """
+
+        icat_relations = mappings.get_icat_relations_for_panosc_non_related_fields(
+            panosc_entity_name,
+        )
+
+        # Remove any duplicate ICAT relations
+        icat_relations = list(dict.fromkeys(icat_relations))
+        if icat_relations:
+            self.filters.append(PythonICATIncludeFilter(icat_relations))
+
+        return icat_relations
+
```

Comment:
This method should not return
```suggestion
```

Reply:
Fixed in 4343a2c67cdba63a0b21e552b7d630eaf0b46cd6